pxtblocks/code-validation/types.ts

```diff
@@ -0,0 +1,35 @@
+namespace pxt.blocks {
+    // A criteria defined in the catalog of all possible criteria for the user to choose from when creating a rubric.
+    export interface CatalogCriteria {
+        id: string; // A unique id (GUID) for the catalog criteria
+        use: string; // Refers to the validator plan this criteria relies upon
+        template: string; // A (mostly) human-readable string describing the criteria. May contain parameters
+        description: string | undefined; // More detailed description
+        docPath: string | undefined; // Path to documentation
+        parameters: CriteriaParameter[] | undefined; // Any parameters that affect the criteria
+    }
+
+    // An instance of a criteria in a rubric.
+    export interface CriteriaInstance {
+        catalogCriteriaId: string;
+        instanceId: string;
+        params: CriteriaParameterValue[] | undefined;
+    }
+
+    // Possible values for CriteriaParameterPicker, these are different ways a user may enter parameter values.
+    export type CriteriaParameterPicker = "blocksPicker" | "numericInput";
```

Comment:
> "numericInput"

I don't think we need a picker for numbers. We can make the param value directly editable.